### PR TITLE
Resolving Discovery request repositry failure

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Link/LinkPatientController.cs  1
         

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  17
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4873278&bid=16154324)

File: src/In.ProjectEKA.DefaultHip/Discovery/DiscoveryRequestRepository.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using System.Linq;
```
Comment (chethandeshpande) [inline src/In.ProjectEKA.DefaultHip/Discovery/DiscoveryRequestRepository.cs]: Imports should be moved inside namespace

File: src/In.ProjectEKA.HipService/Link/LinkPatientController.cs [hunk 2]
```diff
@@ -40,8 +50,9 @@ public LinkPatientController(ILink linkPatient)
             [FromBody] PatientLinkRequest patientLinkRequest)
         {
             var (patientLinkResponse, error) = await linkPatient
-                .VerifyAndLinkCareContext(new HipLibrary.Patient.Model.Request.PatientLinkRequest(patientLinkRequest.Token, linkReferenceNumber));
-        return error != null ? ReturnServerResponse(error) : Ok(patientLinkResponse);
```
Comment (chethandeshpande) [inline src/In.ProjectEKA.HipService/Link/LinkPatientController.cs]: HipLibrary.Patient.Model.Request.PatientLinkRequest add a named import for this

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Link/LinkPatientController.cs  1
         

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  17
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4873278&bid=16154324)

File: test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs [hunk 3]
```diff
@@ -105,9 +112,9 @@ private async void CheckOtpGenerationError()
         {
             var faker = TestBuilder.Faker();
             var consentManagerId = faker.Random.Hash();
-            var linkRequest = new PatientLinkRefRequest(faker.Random.Hash()
-                , new LinkLib(faker.Random.Hash(),faker.Random.Hash(),faker.Random.Hash()
-                    , new [] {new CareContext(faker.Random.Hash())}));
+            var linkRequest = new PatientLinkReferenceRequest(faker.Random.Hash()
```
Comment (ganesanarun) [inline test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs]: can you put this in test builders?

File: src/In.ProjectEKA.HipService/Link/LinkPatientController.cs [hunk 4]
```diff
@@ -2,34 +2,45 @@
...
-            [FromBody] HipLibrary.Patient.Model.Request.PatientLinkReferenceRequest request)
+            [FromBody] PatientLinkReferenceRequest request)
         {
-            var patient = new HipLibrary.Patient.Model.Request.Link(
+            var patient = new LinkLib(
                 consentManagerId,
                 request.Patient.ConsentManagerUserId,
                 request.Patient.ReferenceNumber,
                 request.Patient.CareContexts);
+            var discoveryRequestExists = await discoveryRequestRepository.RequestExistsFor(request.TransactionId);
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/Link/LinkPatientController.cs]:
```suggestion
            var doesRequestExists = await discoveryRequestRepository.RequestExistsFor(request.TransactionId);
```

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Link/LinkPatientController.cs  1
         

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  3
         

Clones removed
==============
+ test/In.ProjectEKA.HipServiceTest/Link/Builder/TestBuilder.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4873278&bid=16154324)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Link/LinkPatientController.cs  1
         

Clones added
============
- test/In.ProjectEKA.HipServiceTest/Link/LinkPatientControllerTest.cs  3
         

Clones removed
==============
+ test/In.ProjectEKA.HipServiceTest/Link/Builder/TestBuilder.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4873278&bid=16154324)

**Pull request merged**